### PR TITLE
#7 Create a module for rewards

### LINKED ISSUE #7: Create a module for rewards management
ℹ️ This issue is part of Epic #2 

## Objective

Create a module for the management of rewards.

## Checklist

- [x] Create module with the necessary functions

### PR DESCRIPTION
**Pull request by Jenietoc**
## Description

In this PR we created the rewards module and the validations module for the contract.

Fixes #7 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: mtk-automation/src/contract_actions/reward.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+fn get_reward_by_type(env: &Env, r_type: &Symbol) -> i128 {
+    let key = DataKey::Rewards;
+    let rewards: Map<Symbol, i128> = env.storage().get(key).unwrap().unwrap();
+
+    rewards.get(r_type.clone()).unwrap().unwrap()
+}
+
+fn transfer(env: &Env, approval_sign: &Signature, to: &Identifier, amount: &i128) {
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/reward.rs:37]: What do you think about moving this function into a new module called for example `token_operations` where we could group all the functions that interact with the token contract?

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]
I left an small suggestion, apart from that it LGTM!

File: mtk-automation/src/contract_actions/reward.rs (hunk 1 above)
Comment (L-Zuluaga) [reply mtk-automation/src/contract_actions/reward.rs:37]: @Jenietoc we also need to replace the `reward` word with `compensation`, which is a more accurate word for this context

File: mtk-automation/src/contract_actions/datakey.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ pub enum DataKey {
     OrgName,
     TokenId,
     AdminId,
-    Rewards,
+    Compensate,
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/datakey.rs:12]: A better word here could be `Compensations`?

File: mtk-automation/src/contract_actions/compensate.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+//! Compensate Module
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/compensate.rs:1]: I'm not sure if  `Compensations` could be a better name for this Module? 🤔 
What do you think @einerzg @Jenietoc?

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]
Left a couple of comments

File: mtk-automation/src/contract_actions/datakey.rs (hunk 2 above)
Comment (Jenietoc) [reply mtk-automation/src/contract_actions/datakey.rs:12]: I can only use 10 characters :sweat_smile:

File: mtk-automation/src/contract_actions/compensate.rs (hunk 3 above)
Comment (Jenietoc) [reply mtk-automation/src/contract_actions/compensate.rs:1]: :eyes: :eyes:

File: mtk-automation/src/contract_actions/datakey.rs (hunk 2 above)
Comment (L-Zuluaga) [reply mtk-automation/src/contract_actions/datakey.rs:12]: Hmm thats true, then I think it makes sense to keep it like this, anyway I'll think on a different word to refer to this part 🤔

Comment (Odraxs) [reply mtk-automation/src/contract_actions/datakey.rs:12]: @L-Zuluaga @einerzg WDYT about renaming `Compensations` to `Perks` ? :thinking:

Comment (einerzg) [reply mtk-automation/src/contract_actions/datakey.rs:12]: @Odraxs I think `Perk` does not represent what is really being done, I prefer `Payments`.

**Review by L-Zuluaga — APPROVED** [review]

**Review by einerzg — APPROVED** [review]

**Pull request merged**